### ISSUE TITLE
OAuth Advice

### ISSUE THREAD

**Issue by lizafloyd** [ga-wdi-exercises/project3#281]
Gentlemen,

We (Andres, Latoya, and I) are embroiled in a oAuth conundrum.  We are hoping to build a networking event manager, and were hoping to set up a convenient, few-click way to add folks on LinkedIn from within our app.  It's looking like oAuth becomes important there, though we could do a less awesome version that doesn't use authorization at all.

Andres seems to have figured out how to get keys (see below), but we wanted to reach out to you guys to see if there's anything we don't know yet that we should take into consideration moving forward.

Here's Andres' code:

You can send them this! I created an app and was able to get this keys!

a-espinoza [7:13 PM]  
ant this is in theory the script that we should use! (found on the "JS SDK" )
<script type="text/javascript" src="//platform.linkedin.com/in.js">
   api_key:   [API_KEY]
   onLoad:    [ONLOAD]
   authorize: [AUTHORIZE]
   lang:      [LANG_LOCALE]
</script

![](https://files.slack.com/files-pri/T2D6FU4JY-F2W3Z4F8R/linkedin.jpg)

Thanks for the guidance!

**Comment by lizafloyd** [comment]
![](https://files.slack.com/files-pri/T2D6FU4JY-F2W3Z4F8R/linkedin.jpg)

**Comment by lizafloyd** [comment]
![image](https://cloud.githubusercontent.com/assets/20960160/19840982/b40a0ffa-9ed8-11e6-8f2d-7a7e7c1b7a59.png)

**Comment by lizafloyd** [comment]
Not sure the code went through.

<script type="text/javascript" src="//platform.linkedin.com/in.js">
   api_key:   [API_KEY]
   onLoad:    [ONLOAD]
   authorize: [AUTHORIZE]
   lang:      [LANG_LOCALE]
</script>

**Comment by lizafloyd** [comment]
Ew, why does it keep deleting the code?? The below is supposed to go in a script tag. Sorry for the technical difficulties.

```
src="//platform.linkedin.com/in.js">
   api_key:   [API_KEY]
   onLoad:    [ONLOAD]
   authorize: [AUTHORIZE]
   lang:      [LANG_LOCALE]
```

**Comment by jshawl** [comment]
try it out! are there any security concerns about having your secret api key visible here and in your app? i.e. does it need to be obscured in rails?

**Comment by lizafloyd** [comment]
I believe it does. LinkedIn recommends it at least. Should I be googling
that? How to obscure an API key in rails?

**Comment by jshawl** [comment]
yes! also check out the figaro gem

**Comment by lizafloyd** [comment]
Awesome! Thanks Jesse!

**Comment by lizafloyd** [comment]
Jesse!

We are looking at the documentation. Would JS SDK work for our purposes, or do you recommend RESTful API calls?

Thanks!

**Comment by amaseda** [comment]
What is the documentation for - LinkedIn API?

If this has to do with your own API, RESTful API calls sounds right though.

**Comment by lizafloyd** [comment]
https://developer.linkedin.com/docs/getting-started-js-sdk

No, we are discussing the linkedin API. We are using oAuth to get user data from the API, and want to make sure the JS SDK format will work. Seems to be working so far, just double checking with you.

**Comment by amaseda** [comment]
Yeah if you're doing this all from the front-end then proceed with the JS SDK!

**Comment by nolds9** [comment]
So just to double check, if you plan on using user info at all later on in your data model, i.e. you need to relate Users to anything in your model, you are going to want to make the calls via Rails...

**Comment by lizafloyd** [comment]
Sorry, Nick, you lost us. Can we do a bit of eli5 here?

What do you mean by model? Users is already a model.....

**Comment by nolds9** [comment]
How are you populating user data for that model?

**Comment by lizafloyd** [comment]
We have not set it up yet, but we are hoping to pull from the linkedin api, mixed with user input.

![](https://files.slack.com/files-pri/T2D6FU4JY-F2WBK8CR3/pasted_image_at_2016_10_31_11_47_am.png)

**Comment by lizafloyd** [comment]
https://wdi-dc.slack.com/files/a-espinoza/F2WBK8CR3/pasted_image_at_2016_10_31_11_47_am.png

**Comment by jshawl** [comment]
can you take a screenshot, drag and drop into this issue?

**Comment by lizafloyd** [comment]
![image](https://cloud.githubusercontent.com/assets/20960160/19860986/378200bc-9f61-11e6-9684-a9d5a6c144fe.png)

**Comment by jshawl** [comment]
I say trust your gut! What's the worst that could happen?

**Comment by nolds9** [comment]
Going to go ahead and close this for now, feel free to reopen / open a new one if you run into any errors
